R/export.R

```diff
@@ -0,0 +1,149 @@
+##' Export packets as a zip file.
+##'
+##' The packets can be imported into a different repository using the
+##' [orderly2::orderly_import_zip] function.
+##'
+##' This is useful as one-time way to publish your results, for example as an
+##' artefact accompanying a paper. For back-and-forth collaboration, a shared
+##' location should be priviledged.
+##'
+##' @param path the path where the zip file will be created
+##'
+##' @param packets One or more packets to export
+##'
+##' @inheritParams orderly_metadata
+##'
+##' @return Nothing
+##' @export
+orderly_export_zip <- function(path, packets, root = NULL, locate = TRUE) {
+  root <- root_open(root, locate = locate, require_orderly = FALSE,
+                    call = environment())
+
+  index <- root$index$data()
+  packets <- find_all_dependencies(packets, index$metadata)
```

Comment:
I wonder if we should offer a non-recursive option here at some (future) point? So that you can dump out a particular packet but _not_ all the bits that lead to it? This would be much smaller often, and more relevant to most users.

The issue here is this requirement:

> She has a sequence of Orderly reports. The first report pulls in from a private data source (which cannot be shared), does a bit of analysis and produces an artefact. The subsequent reports build from the first one's output and apply further analyses.

So by not including all dependencies then we can be sure that no private data has been exposed. 

At present we need to include all metadata I believe, but importing a packet into a location without its dependent files should be fine so long as the user has `require_complete_tree` set to `FALSE`

Reply:
I hadn't considered the fact that orderly might be storing the inputs to the first task. I think she is using `orderly_shared_resource` to ingest the data, which is probably going to leak into the zip file.

On the other hand, not including the first packet means you can't run the reproduce the second one, at least not without some shenanigans to extract the inputs from the second packet.

I think we might need a way of including the first packet, but only its artefacts, not its inputs.

I'll check with Sangeeta exactly how she'd doing it to make sure nothing is leaking accidentally.

Reply:
Nevermind, I was wrong about where the data comes from. It actually comes from an Microsoft Access database, and the first report opens up a connection and retrieves the data it needs. There shouldn't be any particular concern about data leakage, since anything that is fetched from the database can be made public.

We might need to do something different in the future, but this seems fine for now.